crates/viewer/re_viewer/src/background_tasks.rs

```diff
@@ -60,7 +60,10 @@ impl BackgroundTasks {
         self.promises
             .remove(name.as_ref())
             .and_then(|promise| match promise.try_take() {
-                Ok(any) => Some(*any.downcast::<T>().unwrap()),
+                Ok(any) => Some(
+                    *any.downcast::<T>()
+                        .unwrap_or_else(|err| panic!("downcast failure: {err:?}")),
```

Comment:
I think this is an example where a panic is triggered _not_ by an internal error but by an issue with the calling code.

Reply:
I am thinking it may be ideal to update the API to return a promise or result with a promise so that the calling code can poll it without the need for this downcast. We could add a separate API function to do this without breaking any API users. WDYT?